saffron/e2e-test.sh

```diff
@@ -61,16 +85,28 @@ if ! cargo run --release --bin saffron decode -i "$ENCODED_FILE" -o "$DECODED_FI
     exit 1
 fi
 
-# Compare files
-echo "Comparing original and decoded files..."
-if cmp -s "$INPUT_FILE" "$DECODED_FILE"; then
-    echo "✓ Success: Files are identical"
-    echo "Cleaning up temporary files..."
-    rm -f "$ENCODED_FILE" "$DECODED_FILE"
-    exit 0
-else
-    echo "✗ Error: Files differ"
-    echo "Keeping temporary files for inspection"
-    diff "$INPUT_FILE" "$DECODED_FILE"
+# Compare file to original
+compare_files "$INPUT_FILE" "$DECODED_FILE"
+
+# Usage example in your script:
+perturb_bytes "$INPUT_FILE" "$PERTURBED_FILE" 0.1
+
+echo "Calculating diff for upated $INPUT_FILE (stored updated data in $PERTURBED_FILE)"
+cargo run --release --bin saffron calculate-diff --old "$INPUT_FILE" --new "$PERTURBED_FILE" -o "$ENCODED_DIFF_FILE" $SRS_ARG
```

Comment:
```suggestion
cargo run --release --bin saffron calculate-diff --old "$INPUT_FILE" --new "$PERTURBED_FILE" -o "$ENCODED_DIFF_FILE" "$SRS_ARG"
```

(and the others below, to make shellcheck happy)